src/ACMESharp/Crypto/JOSE/IJwsSigner.cs

```diff
@@ -7,18 +7,16 @@ namespace ACMESharp.Crypto.JOSE
     /// Defines the interface for a tool that provides the required
     /// JOSE Web Signature (JWS) functions as used by the ACME protocol.
     /// </summary>
-    public interface IJwsTool : IDisposable
+    public interface IJwsSigner : IDisposable
     {
         string JwsAlg
         { get; }
 
-        void Init();
+        string ExportPrivateJwk();
```

Comment:
You renamed `ExportJwsk` to `ExportPublicJwk` which is right on, since that exports only the public component of a key pair as per the various [JWK specs](https://tools.ietf.org/html/rfc7517#appendix-A), it is more accurate.

But here you also renamed the general-purpose `Export` method to `ExportPrivateJwk` which is not accurate.  In the RSA and ECDSA implementations, this produces an XML export of the entire key pair.  That's in keeping with the intention of the original method, to produce a serialized form of the key that can be used to later constitute it, but it is definitely not a JWK representation.

I would  recommend restoring the original name.  Right now there is no need for a JWK export of the private key component, so I wouldn't go down that route unless you have some other intentions?

Reply:
I agree. I wanted to pronounce the fact, that this will export the private key _also_, but currently it seems to export the private key only (which is neither correct nor intended).